client/src/redux/index.js

```diff
@@ -383,6 +384,13 @@ export const userFetchStateSelector = state => state[MainApp].userFetchState;
 export const userProfileFetchStateSelector = state =>
   state[MainApp].userProfileFetchState;
 export const usernameSelector = state => state[MainApp].appUsername;
+export const usernameDisplaySelector = state => {
+  const username = usernameSelector(state);
+  return (
+    state[MainApp].usernameDisplay ||
+    state[MainApp].user[username]?.usernameDisplay
```

Comment:
Now we have `state.appUsername`, `state.usernameDisplay`, `state.user[username].username` and `state.user[username].usernameDisplay`.  We don't need this complexity, since only one user can log in at once.

I think we should take a step back, fix that and then revisit this.

As far as I can see the fix is just to delete `appUsername` and have a single `state.user` that stores what `state.user[username]` does currently.

Reply:
I am happy to delete `appUsername`, because I do not understand it.

> have a single state.user that stores what state.user[username] does currently.

Are you saying instead of:
```
user: {
  oliver: { username: oliver, usernameDisplay: Oliver }
}
```
This:
```
user: {
  username: oliver,
  usernameDisplay: Oliver
}
```

Reply:
Yep, exactly that. I can't think of a purpose for it either. 

I'm not sure how much disentangling it will require, but it would be safest to do it via a separate PR.

Reply:
One last clarification?:
>do it via a separate PR.

This PR:
- Refactor from `user: { username: {}}` to `user: { }`

Another PR:
- Remove `appUsername`

Reply:
I'd say: 

PR number 1 - remove appUsername and `user: { username: {}}` to `user: { }`.
PR number 2 - this PR

As far as I can see there are two distinct goals 1) refactor redux 2) make the client use `usernameDisplay` correctly. I wouldn't mix the two up.